src/book/02-system/01-nubots/03-continuous-integration.mdx

```diff
@@ -0,0 +1,134 @@
+4. Agents receive a job, clone the code, pull the existing `master` image for DockerHub, and build any new Docker changes. They then mount the code and run the job in a container based on the image, reporting the results back to Buildkite.
+5. Buildkite receives the results and updates the build status in the Buildkite UI as well as on the PR page on GitHub.
+6. When the build passes, the pull request is reviewed by another member of the team and merged into `master` only when approved.
+7. Merging into `master` then triggers another CI build. This time, in addition to building the code, a separate job is created to rebuild the master image and push it to DockerHub.
+
+<TabbedImages>
+
+!['Diagram of CI Overview for pull requests'](./images/ci-overview-pr.svg 'CI Overview PR')
+
+!['Diagram of CI Overview for master'](./images/ci-overview-master.svg 'CI Overview Master')
+
+</TabbedImages>
+
+## The build pipeline
+
+The build pipeline is configured in the [`.buildkite/pipeline.yml`](https://github.com/NUbots/NUbots/blob/master/.buildkite/pipeline.yml) file in the NUbots repo. This file specifies a number of "build steps", which are separate jobs such as "Build for NUC" and "Validate C++ and Protobuf formatting". We edit this file to add, remove, or change build jobs.
+
+The [Buildkite pipeline documentation](https://buildkite.com/docs/pipelines) has everything you need to know to make changes.
+
+## Monitoring and troubleshooting builds
+
+Buildkite has live-updating logs of each build, which you can use to monitor the build progress or troubleshoot errors when a build fails.
+
+If for example your PR build fails due to formatting errors, you can check the logs to see which files are incorrectly formatted. To fix the build, you would reformat the offending files and push a new commit, which will trigger a rebuild.
+
+To view the logs for a build, scroll to the build checks at the bottom of the PR page and click the **Details** link on the `buildkite/nubots/pr` check.
+
+![Screenshot of the build checks section of the PR page, with arrow pointing to primary link](./images/ci-build-logs.png 'Link to the build details with logs')
+
+## Setting up a new agent (Ubuntu)
+
+The following instructions show how to setup and configure a new Buildkite agent with Docker and Docker Compose. These steps have been collected into a [single script you can download here](assets/ci_setup.sh). Before running the script, ensure you have set the Buildkite agent token and DockerHub password correctly in the env variables at the top of the script. Also, make sure to run the script with `sudo`.
+
+<Alert type="warning">
+
+**Note:** the final step below, [Install Docker Compose](#install-docker-compose), is not in the script, and must be done manually in order to install the latest version.
```

Comment:
Why not just put it in the script? I assume it can just be installed via pip to get the latest version

Reply:
The instructions on the website have the latest version hard-coded in the install command. Didn't know it could be installed via pip: https://docs.docker.com/compose/install/#install-using-pip